libs/core/coroutines/include/hpx/coroutines/detail/context_base.hpp

```diff
@@ -208,7 +215,7 @@ namespace hpx { namespace threads { namespace coroutines { namespace detail {
         ~context_base() noexcept
         {
             HPX_ASSERT(!running());
-            HPX_ASSERT(exited());
+            HPX_ASSERT(exited() || m_phase == 0);
```

Comment:
Can you explain the rationale of this change, please?

Reply:
This and the other changes to the assertions in this file were an attempt to loosen the pre- and post-conditions enough that we can eagerly initialize the stack on a thread that hasn't been used yet, but still keep most of the sanity checks that we have in place.

This particular change is because with the eagerly initialized threads it's possible to have a thread that is initialized but never used. In that case the thread would have `m_state = `ctx_ready` when the thread is destructed, which is fine. A more accurate assertion would probably be `HPX_ASSERT(exited() || (ready() && m_phase == 0))`, i.e. at thread destruction a thread should:
- not be running, and
- have run and exited, or
- be ready but not have run at all (i.e. `m_phase == 0`)

Reply:
I've updated the assertion.